webroot/js/components/chat/chat.js

```diff
@@ -164,21 +166,39 @@ export default class Chat extends Component {
 
   render(props, state) {
     const { username, messagesOnly, chatInputEnabled } = props;
-    const { messages, inputEnabled, chatUserNames } = state;
-
-    const messageList = messages.map((message) => (html`<${Message} message=${message} username=${username} key=${message.id} />`));
+    const { messages, chatUserNames } = state;
+
+    const messageList = messages.map(
+      (message) =>
+        html`<${Message}
+          message=${message}
+          username=${username}
+          key=${message.id}
+        />`
+    );
+
+    // After the render completes (based on requestAnimationFrame) then jump to bottom.
+    // This hopefully fixes the race conditions where jumpTobottom fires before the
+    // DOM element has re-drawn with its new size.
+    if (this.jumpToBottomPending) {
+      this.jumpToBottomPending = false;
+      window.requestAnimationFrame(
+        function () {
+          jumpToBottom(this.scrollableMessagesContainer.current);
```

Comment:
we could do `this.jumpToBottom = this.jumpToBottom.bind(this);` in the constructor, for consistency.